### PR TITLE
Cleanup/refactor on `CLI::Media#remove` command

### PR DESCRIPTION
**Pull request by mjankowski**
Mostly spec coverage. Includes slight refactor CLI::Media while I was in there.

### CODE REVIEW

File: lib/mastodon/cli/media.rb
Comment (ClearlyClaire) [inline lib/mastodon/cli/media.rb:1]: I am honestly unsure about the refactor, can I suggest splitting it from the test coverage changes?

Comment (mjankowski) [reply lib/mastodon/cli/media.rb:1]: Sure, pulled the spec commits into https://github.com/mastodon/mastodon/pull/27437 - will rebase this one for further review assuming that one is merged.

Comment (mjankowski) [reply lib/mastodon/cli/media.rb:1]: Rebased this against main after the specs were merged - so this is now just a refactor. Updated title to reflect that.

This maybe more straightforward to review commit by commit - I tried to keep each one as a logical somewhat atomic change. If you have feedback on or objection to a specific commit or change, I could hypothetically back just that out but leave the rest.

**Comment by mjankowski** [comment]
This is one of my oldest PRs where I think there's still good potential for cleanup but where we're at a review impasse. I'm going to attempt to pull out what I think are the remaining useful pieces into smaller PRs, rebasing this one if/when merged, and see how much of it is useful, and then close this one when done.

**Comment by mjankowski** [comment]
Closing as useful portions have been extracted. May re-attempt some of this in other ways in future.

**Pull request closed without merging**